src/components/StatusTag/StatusTag.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import PropTypes from 'prop-types';
+import { STATUS } from '@/utils/constants';
+
+const StatusTag = ({ as, children, padding, color, bgColor, width, fontSize, borderRadius }) => {
+  return (
+    <Wrapper>
+      <StyledDiv
+        padding={padding}
+        bgColor={bgColor}
+        color={color}
+        width={width}
+        fontSize={fontSize}
+        borderRadius={borderRadius}>
+        {decideText(as, bgColor, children)}
+      </StyledDiv>
+    </Wrapper>
+  );
+};
+
+const decideColor = ({ as, bgColor, theme }) => {
+  if (bgColor) {
+    return bgColor;
+  }
+
+  if (as) {
+    return theme.colors[STATUS[as]];
+  }
+
+  return theme.colors.lighterGray;
+};
+
+const decideText = (as, bgColor, children) => {
+  if (bgColor && children) {
+    return children;
+  }
+
+  if ((bgColor && !children) || (!bgColor && children)) return '배경색과 텍스트 둘 다 입력해주세요';
+
+  if (as) {
+    return STATUS[as];
+  }
+
+  return '내용을 입력해 주세요';
+};
```

Comment:
children으로 들어가는 것이 사용자에게 보여주는 메시지인가요? 개발자가 보는 메시지인가요?
아래 같이 써도 동작은 하는데 어떤 부분을 염두에 두신 건지 궁금합니다

```
      <StyledDiv>
         {children || (as && decideText(as))}
      </StyledDiv>
```

```suggestion
const decideText = (as) => {
  return STATUS[as];
};
```

Reply:
as, bgColor, children을 입력받을 수 있는데, 그렇게 되면 총 경우의 수가 8가지가 됩니다.

이중 bgColor와 children중 적어도 하나를 사용했다는 것은 label을 커스톰하겠다는 의미로 처리하기 위해, 둘 중 하나만 입력하는 경우, 화면에 엉뚱한 텍스트를 출력하여 개발자에게 알려주기 위함입니다.